src/executor/query/GetEdgesExecutor.cpp

```diff
@@ -11,71 +11,109 @@
+    auto kind = ge_->dataKind();
+    if (kind == GetEdges::DataKind::kEdge) {
+        status = buildEdgeRequestDataSet();
+    } else {
+        status = buildPathRequestDataSet();
+    }
+    NG_RETURN_IF_ERROR(status);
     return getEdges();
 }
 
-folly::Future<Status> GetEdgesExecutor::getEdges() {
-    SCOPED_TIMER(&execTime_);
-
-    GraphStorageClient *client = qctx()->getStorageClient();
+Status GetEdgesExecutor::close() {
+    // clear the members
+    reqDs_.rows.clear();
+    return Executor::close();
+}
 
-    auto *ge = asNode<GetEdges>(node());
-    nebula::DataSet edges({kSrc, kType, kRank, kDst});
-    const auto& spaceInfo = qctx()->rctx()->session()->space();
-    if (ge->src() != nullptr &&
-        ge->type() != nullptr &&
-        ge->ranking() != nullptr &&
-        ge->dst() != nullptr) {
+Status GetEdgesExecutor::buildEdgeRequestDataSet() {
+    SCOPED_TIMER(&execTime_);
+    const auto &spaceInfo = qctx()->rctx()->session()->space();
+    auto inputVar = ge_->inputVar();
+    VLOG(1) << "GetEdges Input : " << inputVar;
+    if (ge_->src() != nullptr && ge_->type() != nullptr && ge_->ranking() != nullptr &&
+        ge_->dst() != nullptr) {
         // Accept Table such as | $a | $b | $c | $d |... which indicate src, ranking or dst
-        auto valueIter = ectx_->getResult(ge->inputVar()).iter();
-        auto expCtx = QueryExpressionContext(qctx()->ectx());
-        for (; valueIter->valid(); valueIter->next()) {
-            auto src = ge->src()->eval(expCtx(valueIter.get()));
-            auto type = ge->type()->eval(expCtx(valueIter.get()));
-            auto ranking = ge->ranking()->eval(expCtx(valueIter.get()));
-            auto dst = ge->dst()->eval(expCtx(valueIter.get()));
-            if (!SchemaUtil::isValidVid(src, spaceInfo.spaceDesc.vid_type)
-                    || !SchemaUtil::isValidVid(dst, spaceInfo.spaceDesc.vid_type)
-                    || !type.isInt() || !ranking.isInt()) {
+        auto iter = ectx_->getResult(inputVar).iter();
+        QueryExpressionContext ctx(ectx_);
+        for (; iter->valid(); iter->next()) {
+            auto src = ge_->src()->eval(ctx(iter.get()));
+            auto type = ge_->type()->eval(ctx(iter.get()));
+            auto ranking = ge_->ranking()->eval(ctx(iter.get()));
+            auto dst = ge_->dst()->eval(ctx(iter.get()));
+            if (!SchemaUtil::isValidVid(src, spaceInfo.spaceDesc.vid_type) ||
+                !SchemaUtil::isValidVid(dst, spaceInfo.spaceDesc.vid_type) || !type.isInt() ||
+                !ranking.isInt()) {
                 LOG(WARNING) << "Mismatched edge key type";
                 continue;
             }
-            edges.emplace_back(Row({
-                std::move(src), type, ranking, std::move(dst)
-            }));
+            reqDs_.emplace_back(Row({std::move(src), type, ranking, std::move(dst)}));
+        }
+    }
+    return Status::OK();
+}
+
+Status GetEdgesExecutor::buildPathRequestDataSet() {
```

Comment:
I thinks you could do this when generate the plan instead of executor for this will lead the executor to complicate.